src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/SPARQLStar2CypherTranslatorImpl.java

```diff
@@ -806,4 +807,85 @@ protected static CypherQuery getVarVarVar(final Node s, final Node p, final Node
                         .add(new AliasedExpression(new GetItemExpression(a10, 1), gen.getRetVar(o)))
                         .build());
     }
+
+    @Override
+    public CypherMatchQuery rewriteJoins(final CypherMatchQuery query) {
+        final CypherQueryBuilder builder = new CypherQueryBuilder();
+        final List<BooleanCypherExpression> variableJoins = query.getConditions().stream()
+                .filter(x -> x instanceof EqualityExpression
+                        && ((EqualityExpression) x).getLeftExpression() instanceof CypherVar
+                        && ((EqualityExpression) x).getRightExpression() instanceof CypherVar
+                        && query.getMatchVars().contains((CypherVar) ((EqualityExpression) x).getLeftExpression())
+                        && query.getMatchVars().contains((CypherVar) ((EqualityExpression) x).getRightExpression()))
+                .collect(Collectors.toList());
```

Comment:
I assume that there can be cases in which the `variableJoins` list created by this statement is empty. In such cases, there is no need to go through the rest of the code. Instead, simply return the given Match query.